### PR TITLE
feat: add `try_send_raw` and `try_receive`

### LINKED ISSUE #32: Add try_receive method
`.receive()` polls for new input internally. A `.try_receive()` method would allow the OS to handle the polling itself and do other useful work between two polls rather than hang until the next input byte or (if supported by the OS) the current kernel thread gets preempted.

### PR DESCRIPTION
**Pull request by mkroening**
Closes https://github.com/rust-osdev/uart_16550/issues/32.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -62,12 +62,17 @@
 #![warn(missing_docs)]
 #![cfg_attr(docsrs, feature(doc_cfg))]
 
+use core::fmt;
+
 use bitflags::bitflags;
 
-macro_rules! wait_for {
+macro_rules! spin_unwrap {
```
Comment (phil-opp) [inline src/lib.rs]: I'm not sure about the `unwrap` name. Maybe something like `retry_until_ok`?

**Review by phil-opp — APPROVED** [review]
Thanks a lot, this is very useful!

File: src/lib.rs (hunk 1 above)
Comment (mkroening) [reply src/lib.rs]: Done 👍

**Comment by phil-opp** [comment]
Published as v0.3.1

**Pull request merged**